{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "c:\\Python310\\lib\\site-packages\\google\\auth\\_default.py:78: UserWarning: Your application has authenticated using end user credentials from Google Cloud SDK without a quota project. You might receive a \"quota exceeded\" or \"API not enabled\" error. See the following page for troubleshooting: https://cloud.google.com/docs/authentication/adc-troubleshooting/user-creds. \n",
      "  warnings.warn(_CLOUD_SDK_CREDENTIALS_WARNING)\n"
     ]
    }
   ],
   "source": [
    "from google.cloud import bigquery\n",
    "\n",
    "# Construct a BigQuery client object.\n",
    "client = bigquery.Client()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "def list_diseases():\n",
    "    query = \"SELECT DISTINCT trait_reported FROM `thegwensproject.genomiX.disease_variant_gene` order by trait_reported;\"\n",
    "    query_job = client.query(query)  # Make an API request.\n",
    "    return query_job.to_dataframe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def count_variants(chr, pos):\n",
    "    if pos == 0:\n",
    "        query = (\n",
    "            \"SELECT chr_id_b37, position_b37, COUNT(*) as count FROM thegwensproject.genomiX.variants WHERE chr_id_b37 = '\" +str(chr) +\"' GROUP BY chr_id_b37, position_b37 HAVING COUNT(*) > 1 LIMIT 20;\")\n",
    "    else:\n",
    "        query = (\n",
    "            \"SELECT chr_id_b37, position_b37, COUNT(*) as count FROM thegwensproject.genomiX.variants WHERE chr_id_b37 = '\" +str(chr) +\"' AND position_b37 = \" + str(pos) + \" GROUP BY chr_id_b37, position_b37 HAVING COUNT(*) > 1 LIMIT 20;\")\n",
    "    return query\n",
    "query_job = client.query(count_variants(6, 166956723))\n",
    "for row in query_job:\n",
    "    # Row values can be accessed by field name or index.\n",
    "    print(row)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def variant_info(chr, pos):\n",
    "    query = (\n",
    "        \"SELECT * FROM thegwensproject.genomiX.variants WHERE chr_id_b37 = '\" +str(chr) +\"' AND position_b37 = \" + str(pos) + \" LIMIT 20;\")\n",
    "    return query\n",
    "\n",
    "query_job = client.query(variant_info(6, 166956723))\n",
    "for row in query_job:\n",
    "    # Row values can be accessed by field name or index.\n",
    "    print(row[9])\n",
    "    print (row)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def tag_disease_variant_info(chr, pos):\n",
    "    query = (\n",
    "        \"SELECT * FROM thegwensproject.genomiX.disease_variant_gene WHERE tag_chrom = '\" +str(chr) +\"' AND tag_pos = \"+ str(pos) +\"  LIMIT 20;\")\n",
    "    return query\n",
    "\n",
    "query_job = client.query(tag_disease_variant_info(6, 166980468))\n",
    "for row in query_job:\n",
    "    # Row values can be accessed by field name or index.\n",
    "    print(row['lead_pos'])\n",
    "    print (row)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "def lead_disease_variant_info(chr, pos):\n",
    "    query = (\n",
    "        \"SELECT * FROM thegwensproject.genomiX.disease_variant_gene WHERE lead_chrom = '\" +str(chr) +\"' AND lead_pos = \"+ str(pos) +\"  LIMIT 20;\")\n",
    "    return query\n",
    "query_job = client.query(lead_disease_variant_info(6, 166956723))\n",
    "for row in query_job:\n",
    "    # Row values can be accessed by field name or index.\n",
    "    print (row)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def exact_disease_variant_info(leadchr, leadpos, tagchr, tagpos):\n",
    "    query = ( \"SELECT * FROM thegwensproject.genomiX.disease_variant_gene WHERE lead_chrom = '\" +str(leadchr) +\"' AND lead_pos = \"+ str(leadpos) +\" AND tag_chrom = '\" +str(tagchr) +\"' AND tag_pos = \"+ str(tagpos) +\"  LIMIT 20;\")\n",
    "    return query\n",
    "query_job = client.query(exact_disease_variant_info(6, 166980468, 6, 166952906))\n",
    "for row in query_job:\n",
    "    # Row values can be accessed by field name or index.\n",
    "    print (row)\n",
    "    print (row['trait_reported'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def lead_disease_info(chr, pos):\n",
    "    query = (\n",
    "        \"SELECT * FROM thegwensproject.genomiX.variant_disease WHERE lead_chrom = '\" +str(chr) +\"' AND lead_pos = \"+ str(pos) +\"  LIMIT 20;\")\n",
    "    return query\n",
    "query_job = client.query(lead_disease_info(6, 166956723))\n",
    "for row in query_job:\n",
    "    # Row values can be accessed by field name or index.\n",
    "    print (row['trait_reported'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def tag_disease_info(chr, pos):\n",
    "    query = (\n",
    "        \"SELECT * FROM thegwensproject.genomiX.variant_disease WHERE tag_chrom = '\" +str(chr) +\"' AND tag_pos = \"+ str(pos) +\"  LIMIT 20;\")\n",
    "    return query\n",
    "query_job = client.query(tag_disease_info(6, 166952906))\n",
    "for row in query_job:\n",
    "    # Row values can be accessed by field name or index.\n",
    "    print (row['trait_reported'])\n",
    "    print (row['lead_chrom'] + \" \"+ str(row['lead_pos']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def exact_disease_info(leadchr, leadpos, tagchr, tagpos):\n",
    "    query = (\n",
    "        \"SELECT * FROM thegwensproject.genomiX.variant_disease WHERE tag_chrom = '\" +str(tagchr) +\"' AND tag_pos = \"+ str(tagpos) +\" AND lead_chrom = '\" +str(leadchr) +\"' AND lead_pos = \"+ str(leadpos) +\"  LIMIT 20;\")\n",
    "    return query\n",
    "query_job = client.query(exact_disease_info(6, 166980468, 6, 166952906))\n",
    "for row in query_job:\n",
    "    # Row values can be accessed by field name or index.\n",
    "    print (row['trait_reported'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def disease_catcher(disease):\n",
    "    query = f'SELECT * FROM thegwensproject.genomiX.variant_disease WHERE trait_reported = \"{disease}\" LIMIT 20;'\n",
    "    return query\n",
    "query_job = client.query(disease_catcher(\"Alzheimer's disease\"))\n",
    "for row in query_job:\n",
    "    # Row values can be accessed by field name or index.\n",
    "    print (row['lead_chrom'] + \" \"+ str(row['lead_pos']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#use % of different parameters like acestry, for the disease specific genes.\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "lead = pd.read_csv('lead_chrom_trait.csv')\n",
    "tag = pd.read_csv('tag_chrom_trait.csv')\n",
    "df_lead_pivot = lead.pivot(index='trait_reported', columns='lead_chrom')\n",
    "df_tag_pivot = tag.pivot(index='trait_reported', columns='tag_chrom')\n",
    "df_combined = df_lead_pivot.join(df_tag_pivot, lsuffix='_lead', rsuffix='_tag')\n",
    "\n",
    "df_combined.reset_index(inplace=True)\n",
    "\n",
    "#chrom_trait_joined = pd.merge(lead, tag, on='trait_reported', how='inner')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>trait_reported</th>\n",
       "      <th colspan=\"9\" halign=\"left\">count_lead</th>\n",
       "      <th>...</th>\n",
       "      <th colspan=\"10\" halign=\"left\">count_tag</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>lead_chrom</th>\n",
       "      <th></th>\n",
       "      <th>1</th>\n",
       "      <th>10</th>\n",
       "      <th>11</th>\n",
       "      <th>12</th>\n",
       "      <th>13</th>\n",
       "      <th>14</th>\n",
       "      <th>15</th>\n",
       "      <th>16</th>\n",
       "      <th>17</th>\n",
       "      <th>...</th>\n",
       "      <th>21</th>\n",
       "      <th>22</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>X</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>!Aliquae complicationes praecoces traumatis</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>35.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>\"Abnormal blood-pressure reading, without diag...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1157.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>\"Abnormal findings on diagnostic imaging and i...</td>\n",
       "      <td>208.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>\"Abnormal findings on examination of blood, wi...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>\"Abnormal findings on examination of other bod...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 47 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               trait_reported count_lead  \\\n",
       "lead_chrom                                                             1   \n",
       "0                 !Aliquae complicationes praecoces traumatis        0.0   \n",
       "1           \"Abnormal blood-pressure reading, without diag...        0.0   \n",
       "2           \"Abnormal findings on diagnostic imaging and i...      208.0   \n",
       "3           \"Abnormal findings on examination of blood, wi...        0.0   \n",
       "4           \"Abnormal findings on examination of other bod...        0.0   \n",
       "\n",
       "                                                        ... count_tag       \\\n",
       "lead_chrom      10    11   12   13   14   15   16   17  ...        21   22   \n",
       "0              0.0   0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...       0.0  0.0   \n",
       "1           1157.0   0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...       0.0  0.0   \n",
       "2              0.0   0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...       0.0  0.0   \n",
       "3              0.0   0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...       0.0  0.0   \n",
       "4              0.0  33.0  0.0  0.0  0.0  0.0  0.0  0.0  ...       0.0  0.0   \n",
       "\n",
       "                                                     \n",
       "lead_chrom    3     4    5    6    7    8    9    X  \n",
       "0           0.0  35.0  0.0  0.0  0.0  0.0  0.0  0.0  \n",
       "1           0.0   0.0  0.0  0.0  0.0  0.0  0.0  0.0  \n",
       "2           0.0   0.0  0.0  0.0  0.0  0.0  0.0  0.0  \n",
       "3           0.0   0.0  0.0  0.0  0.0  9.0  0.0  0.0  \n",
       "4           0.0   0.0  0.0  0.0  0.0  0.0  0.0  0.0  \n",
       "\n",
       "[5 rows x 47 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_combined.fillna(0, inplace=True)\n",
    "df_combined.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_67322/3530631317.py:5: PerformanceWarning: dropping on a non-lexsorted multi-index without a level parameter may impact performance.\n",
      "  df_percent.drop('total', axis=1, inplace=True)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>trait_reported</th>\n",
       "      <th colspan=\"9\" halign=\"left\">count_lead</th>\n",
       "      <th>...</th>\n",
       "      <th colspan=\"10\" halign=\"left\">count_tag</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>lead_chrom</th>\n",
       "      <th></th>\n",
       "      <th>1</th>\n",
       "      <th>10</th>\n",
       "      <th>11</th>\n",
       "      <th>12</th>\n",
       "      <th>13</th>\n",
       "      <th>14</th>\n",
       "      <th>15</th>\n",
       "      <th>16</th>\n",
       "      <th>17</th>\n",
       "      <th>...</th>\n",
       "      <th>21</th>\n",
       "      <th>22</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>X</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>!Aliquae complicationes praecoces traumatis</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>50.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>\"Abnormal blood-pressure reading, without diag...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>50.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>\"Abnormal findings on diagnostic imaging and i...</td>\n",
       "      <td>50.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>\"Abnormal findings on examination of blood, wi...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.612903</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>\"Abnormal findings on examination of other bod...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>50.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 47 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               trait_reported count_lead  \\\n",
       "lead_chrom                                                             1   \n",
       "0                 !Aliquae complicationes praecoces traumatis        0.0   \n",
       "1           \"Abnormal blood-pressure reading, without diag...        0.0   \n",
       "2           \"Abnormal findings on diagnostic imaging and i...       50.0   \n",
       "3           \"Abnormal findings on examination of blood, wi...        0.0   \n",
       "4           \"Abnormal findings on examination of other bod...        0.0   \n",
       "\n",
       "                                                      ... count_tag            \\\n",
       "lead_chrom    10    11   12   13   14   15   16   17  ...        21   22    3   \n",
       "0            0.0   0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...       0.0  0.0  0.0   \n",
       "1           50.0   0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...       0.0  0.0  0.0   \n",
       "2            0.0   0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...       0.0  0.0  0.0   \n",
       "3            0.0   0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...       0.0  0.0  0.0   \n",
       "4            0.0  50.0  0.0  0.0  0.0  0.0  0.0  0.0  ...       0.0  0.0  0.0   \n",
       "\n",
       "                                                     \n",
       "lead_chrom     4    5    6    7         8    9    X  \n",
       "0           50.0  0.0  0.0  0.0  0.000000  0.0  0.0  \n",
       "1            0.0  0.0  0.0  0.0  0.000000  0.0  0.0  \n",
       "2            0.0  0.0  0.0  0.0  0.000000  0.0  0.0  \n",
       "3            0.0  0.0  0.0  0.0  1.612903  0.0  0.0  \n",
       "4            0.0  0.0  0.0  0.0  0.000000  0.0  0.0  \n",
       "\n",
       "[5 rows x 47 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_combined['total'] = df_combined.iloc[:,1:].sum(axis=1)\n",
    "df_percent = df_combined.copy()\n",
    "for column in df_percent.columns[1:-1]:\n",
    "    df_percent[column] = df_percent[column] / df_percent['total']*100\n",
    "df_percent.drop('total', axis=1, inplace=True)\n",
    "df_percent.head()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.10"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
